extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -82,6 +82,37 @@ public <T> CommonPanacheQueryImpl<T> project(Class<T> type) {
             throw new PanacheQueryException("Unable to perform a projection on a named query");
         }
 
+        if (query.trim().toLowerCase().startsWith("select new ")) {
+            throw new PanacheQueryException("Unable to perform a projection on a 'select new' query: " + query);
+        }
+
+        // If the query starts with a select clause, we generate an HQL query
+        // using the fields in the select clause:
+        // Initial query: select e.field1, e.field2 from EntityClass e
+        // New query: SELECT new org.acme.ProjectionClass(e.field1, e.field2) from EntityClass e
+        if (query.trim().toLowerCase().startsWith("select ")) {
```

Comment:
Davide :) Please :) My eyes are bleeding from all those `query.trim().toLowerCase()` calls :)

Reply:
Is this better? https://github.com/quarkusio/quarkus/pull/26510/commits/966f6af57bd943905851d7e52bec1d562afff791